.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,84 @@
+name: Trigger image scanning on pull request and changes to pr
+on:
+  pull_request:
+    types: [opened, edited, synchronize]
+
+jobs:
+  deploy:
+    name: Trigger image scanning
+    runs-on: ubuntu-latest
+    env:
+      PEAK_USER_ID: 8877
+      BRANCH: ${{ github.head_ref }}
+      SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      GH_PAGES_BASE_URL: https://peak-ai.github.io/platform-global-images
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get changed directories
+        id: changed-directories
+        uses: tj-actions/changed-files@v23
+        with:
+          files_ignore: |
+            *.md
+            .github/workflows
+          dir_names: true
+
+      - name: List all changed directories/images
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            echo "$directory was changed"
+          done
+
+      - name: Run docker build on changed images
+        run: |
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            tag=$(echo $directory | tr / -)
+            echo "Building $directory"
+            docker build $directory --build-arg PEAK_USER_ID=$PEAK_USER_ID -t $tag
+          done
+
+      - name: Run snyk scan on changed images
+        if: steps.changed-directories.outputs.any_changed == 'true'
+        shell: bash --noprofile --norc {0}
+        id: snyk-scans
+        run: |
+          npx snyk auth $SNYK_TOKEN
+          mkdir -p /tmp/$BRANCH
+          for directory in ${{ steps.changed-directories.outputs.all_changed_files }}; do
+            tag=$(echo $directory | tr / -)
+            npx snyk container test $tag --json >> $tag.json
+            touch /tmp/$BRANCH/$tag.html
```

Comment:
Do we need this?

Reply:
Hey had added this for testing stuff. Have removed this.